examples/multiple_node_versions/BUILD.bazel

```diff
@@ -0,0 +1,125 @@
+load("@bazel_skylib//rules:write_file.bzl", "write_file")
+load("@bazel_skylib//rules:diff_test.bzl", "diff_test")
+load(":defs.bzl", "my_nodejs")
+load("@build_bazel_rules_nodejs//:index.bzl", "nodejs_binary", "nodejs_test")
+
+# Trivial test fixture: a nodejs program that writes to a file
+write_file(
+    name = "js",
+    out = "some.js",
+    content = ["require('fs').writeFileSync(process.argv[2], 'stuff')"],
```

Comment:
To keep from growing to much duplicate test code, lets move this all over to `e2e/core`. It looks like the additional of

```
    if ctx.attr.toolchain:
        toolchain = ctx.attr.toolchain[platform_common.ToolchainInfo].nodeinfo
    else:
        toolchain = ctx.toolchains["@rules_nodejs//nodejs:toolchain_type"].nodeinfo
```
& the new attr are the load bearing changes in `my_nodejs` rule.

I'd make those changes there, bring in multiple node versions in that WORKSPACE and then bring in a test case to test that you can specify a custom toolchain. You'll want to `select` the toolchain so that it still works on multiple platforms (since we won't be relying on bazel selecting the correct one).

Something like

```
my_nodejs(
    name = "run_15",
    out = "thing_15",
    entry_point = "version.js",
    toolchain = select({
        "@bazel_tools//src/conditions:linux_x86_64": "@node15_linux_amd64//:node_toolchain",
        "@bazel_tools//src/conditions:darwin": "@node15_darwin_amd64//:node_toolchain",
        "@bazel_tools//src/conditions:windows": "@node15_windows_amd64//:node_toolchain",
    }),               
)
```
(doesn't cover the M1 case but on an M1 you can run an intel binary in rosetta compat mode)

Reply:
note that the select statement will sadly download toolchains for all three platforms, I don't think it can be helped but you should mention that in a comment here

Reply:
Sounds good, I'll move those things over and add the select statements